src/driver/mongodb/MongoDriver.ts

```diff
@@ -394,4 +420,21 @@ export class MongoDriver implements Driver {
         return `mongodb://${credentialsUrlPart}${this.options.host || "127.0.0.1"}:${this.options.port || "27017"}/${this.options.database}`;
     }
 
+    /**
+     * Build connection options from MongoConnectionOptions
+     */
+    protected buildConnectionOptions(): any {
+        const mongoOptions: any = {};
+
+        for (let index = 0; index < this.validOptionNames.length; index++) {
+            const optionName = this.validOptionNames[index];
+
+            if (optionName in this.options) {
+                mongoOptions[optionName] = (this.options as any)[optionName];
+            }
```

Comment:
need check `this.options.extra` ? https://github.com/typeorm/typeorm/issues/1741